Mathlib/MeasureTheory/Function/LpSeminorm/ChebyshevMarkov.lean

```diff
@@ -15,58 +15,88 @@ open scoped NNReal ENNReal
 
 namespace MeasureTheory
 
-variable {α E : Type*} {m0 : MeasurableSpace α} [NormedAddCommGroup E]
-  {p : ℝ≥0∞} (μ : Measure α) {f : α → E}
+variable {α E ε' : Type*} {m0 : MeasurableSpace α} [NormedAddCommGroup E]
+  [TopologicalSpace ε'] [ContinuousENorm ε']
+  {p : ℝ≥0∞} (μ : Measure α)
 
 theorem pow_mul_meas_ge_le_eLpNorm (hp_ne_zero : p ≠ 0) (hp_ne_top : p ≠ ∞)
-    (hf : AEStronglyMeasurable f μ) (ε : ℝ≥0∞) :
-    (ε * μ { x | ε ≤ (‖f x‖₊ : ℝ≥0∞) ^ p.toReal }) ^ (1 / p.toReal) ≤ eLpNorm f p μ := by
+    {f : α → ε'} (hf : AEStronglyMeasurable f μ) (ε : ℝ≥0∞) :
+    (ε * μ { x | ε ≤ ‖f x‖ₑ ^ p.toReal }) ^ (1 / p.toReal) ≤ eLpNorm f p μ := by
   rw [eLpNorm_eq_lintegral_rpow_enorm hp_ne_zero hp_ne_top]
   gcongr
   exact mul_meas_ge_le_lintegral₀ (hf.enorm.pow_const _) ε
 
 theorem mul_meas_ge_le_pow_eLpNorm (hp_ne_zero : p ≠ 0) (hp_ne_top : p ≠ ∞)
-    (hf : AEStronglyMeasurable f μ) (ε : ℝ≥0∞) :
-    ε * μ { x | ε ≤ (‖f x‖₊ : ℝ≥0∞) ^ p.toReal } ≤ eLpNorm f p μ ^ p.toReal := by
+    {f : α → ε'} (hf : AEStronglyMeasurable f μ) (ε : ℝ≥0∞) :
+    ε * μ { x | ε ≤ ‖f x‖ₑ ^ p.toReal } ≤ eLpNorm f p μ ^ p.toReal := by
   have : 1 / p.toReal * p.toReal = 1 := by
     refine one_div_mul_cancel ?_
     rw [Ne, ENNReal.toReal_eq_zero_iff]
     exact not_or_intro hp_ne_zero hp_ne_top
-  rw [← ENNReal.rpow_one (ε * μ { x | ε ≤ (‖f x‖₊ : ℝ≥0∞) ^ p.toReal }), ← this, ENNReal.rpow_mul]
+  rw [← ENNReal.rpow_one (ε * μ { x | ε ≤ ‖f x‖ₑ ^ p.toReal }), ← this, ENNReal.rpow_mul]
   gcongr
   exact pow_mul_meas_ge_le_eLpNorm μ hp_ne_zero hp_ne_top hf ε
 
 /-- A version of Chebyshev-Markov's inequality using Lp-norms. -/
 theorem mul_meas_ge_le_pow_eLpNorm' (hp_ne_zero : p ≠ 0) (hp_ne_top : p ≠ ∞)
-    (hf : AEStronglyMeasurable f μ) (ε : ℝ≥0∞) :
-    ε ^ p.toReal * μ { x | ε ≤ ‖f x‖₊ } ≤ eLpNorm f p μ ^ p.toReal := by
+    {f : α → ε'} (hf : AEStronglyMeasurable f μ) (ε : ℝ≥0∞) :
+    ε ^ p.toReal * μ { x | ε ≤ ‖f x‖ₑ } ≤ eLpNorm f p μ ^ p.toReal := by
   convert mul_meas_ge_le_pow_eLpNorm μ hp_ne_zero hp_ne_top hf (ε ^ p.toReal) using 4
   ext x
   rw [ENNReal.rpow_le_rpow_iff (ENNReal.toReal_pos hp_ne_zero hp_ne_top)]
 
-theorem meas_ge_le_mul_pow_eLpNorm (hp_ne_zero : p ≠ 0) (hp_ne_top : p ≠ ∞)
-    (hf : AEStronglyMeasurable f μ) {ε : ℝ≥0∞} (hε : ε ≠ 0) :
-    μ { x | ε ≤ ‖f x‖₊ } ≤ ε⁻¹ ^ p.toReal * eLpNorm f p μ ^ p.toReal := by
+-- TODO: remove its only use in CondExp/Real.lean, then deprecate and remove this
+/-- A version of Chebyshev-Markov's inequality using Lp-norms. -/
+theorem mul_meas_ge_le_pow_eLpNorm'' (hp_ne_zero : p ≠ 0) (hp_ne_top : p ≠ ∞)
+    {f : α → E} (hf : AEStronglyMeasurable f μ) (ε : ℝ≥0∞) :
+    ε ^ p.toReal * μ { x | ε ≤ ‖f x‖₊ } ≤ eLpNorm f p μ ^ p.toReal :=
+  mul_meas_ge_le_pow_eLpNorm' μ hp_ne_zero hp_ne_top hf ε
+
+theorem meas_ge_le_mul_pow_eLpNorm_enorm (hp_ne_zero : p ≠ 0) (hp_ne_top : p ≠ ∞)
+    {f : α → ε'} (hf : AEStronglyMeasurable f μ)
+    {ε : ℝ≥0∞} (hε : ε ≠ 0) (hmeas_top : ε = ∞ → μ {x | ‖f x‖ₑ = ⊤} = 0) :
```

Comment:
```suggestion
    {ε : ℝ≥0∞} (hmeas_top : ε = ∞ → μ {x | ‖f x‖ₑ = ⊤} = 0) :
```
I think this was accidentally still here (also below)

Reply:
It looks like it, but the lemma is actually false if ε=0 and eLpNorm f p μ = 0 (in that case, the right hand side is \infty * 0 = 0, but the left hand side is positive unless \mu = 0). That condition was already present in the non-enorm version.